### PR TITLE
Peeps C++

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

File: src/openrct2/peep/Staff.cpp [hunk 1]
```diff
@@ -1551,9 +1655,11 @@ uint32 staff_get_available_entertainer_costumes()
     init_scenery();
 
     uint32 entertainerCostumes = 0;
-    for (sint32 i = 0; i < 19; i++) {
-        if (window_scenery_tab_entries[i][0] != -1) {
-            rct_scenery_set_entry* scenery_entry = get_scenery_group_entry(i);
+    for (sint32 i = 0; i < 19; i++)
```
Comment (marijnvdwerf) [inline src/openrct2/peep/Staff.cpp]: What's this `19`?

Comment (Gymnasiast) [reply src/openrct2/peep/Staff.cpp]: MAX_SCENERY_GROUP_OBJECTS

File: src/openrct2/peep/Peep.cpp [hunk 2]
```diff
@@ -105,429 +108,229 @@ enum {
...
-    PEEP_ACTION_SPRITE_TYPE_CLAP,
-    PEEP_ACTION_SPRITE_TYPE_DISGUST,
-    PEEP_ACTION_SPRITE_TYPE_DRAW_PICTURE,
-    PEEP_ACTION_SPRITE_TYPE_BEING_WATCHED,
-    PEEP_ACTION_SPRITE_TYPE_WITHDRAW_MONEY
-};
+static uint8 PeepSpecialSpriteToSpriteTypeMap[] = { PEEP_ACTION_SPRITE_TYPE_NONE, PEEP_ACTION_SPRITE_TYPE_HOLD_MAT,
+                                                    PEEP_ACTION_SPRITE_TYPE_STAFF_MOWER };
+
+static uint8 PeepActionToSpriteTypeMap[] = { PEEP_ACTION_SPRITE_TYPE_CHECK_TIME,
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: Bad alignment. Same for above table.

File: src/openrct2/peep/Peep.cpp [hunk 3]
```diff
@@ -758,147 +589,133 @@ static void peep_decide_whether_to_leave_park(rct_peep *peep)
...
+    50,  // SHOP_ITEM_CANDY_APPLE
+    0,   // SHOP_ITEM_TSHIRT
+    80,  // SHOP_ITEM_DONUT
+    90,  // SHOP_ITEM_COFFEE
+    0,   // SHOP_ITEM_EMPTY_CUP
+    170, // SHOP_ITEM_CHICKEN
+    115, // SHOP_ITEM_LEMONADE
+    0,   // SHOP_ITEM_EMPTY_BOX
+    0,   // SHOP_ITEM_EMPTY_BOTTLE
+    0xFF, 0xFF, 0xFF, 0xFF,
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: Either you do put a table on one line or every item on its own line. Do not mix it up.

Comment (Gymnasiast) [reply src/openrct2/peep/Peep.cpp]: Still needs fixing

File: src/openrct2/peep/Peep.cpp [hunk 4]
```diff
@@ -758,147 +796,133 @@ static void peep_decide_whether_to_leave_park(rct_peep *peep)
...
-    PEEP_THOUGHT_TYPE_SCENERY,
-    PEEP_THOUGHT_TYPE_VERY_CLEAN,
-    PEEP_THOUGHT_TYPE_MUSIC,
-    PEEP_THOUGHT_TYPE_WATCHED,
-    PEEP_THOUGHT_TYPE_NOT_HUNGRY,
-    PEEP_THOUGHT_TYPE_NOT_THIRSTY,
-    PEEP_THOUGHT_TYPE_BATHROOM,
-    PEEP_THOUGHT_TYPE_NONE,
-    PEEP_THOUGHT_TYPE_NONE,
+    PEEP_THOUGHT_TYPE_LOST,        PEEP_THOUGHT_TYPE_TIRED,      PEEP_THOUGHT_TYPE_BAD_LITTER, PEEP_THOUGHT_TYPE_HUNGRY,
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: I really prefer if arrays are horizontal _or_ vertical.

**Comment by janisozaur** [comment]
@Gymnasiast rebased + updated formatting. Is this good now? Please update the xcode project.

File: src/openrct2/peep/Staff.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2/peep/Staff.cpp]: @janisozaur Can you replace this?

File: src/openrct2/peep/Peep.cpp [hunk 5]
```diff
@@ -3057,55 +3330,58 @@ static void peep_update_ride_sub_state_2(rct_peep* peep){
...
     ride->cur_num_customers++;
 
     vehicle->friction += peep->var_41;
-    invalidate_sprite_2((rct_sprite*)vehicle);
+    invalidate_sprite_2((rct_sprite *)vehicle);
 
-    invalidate_sprite_2((rct_sprite*)peep);
-    sprite_move(0x8000, 0, 0, (rct_sprite*)peep);
+    invalidate_sprite_2((rct_sprite *)peep);
+    sprite_move(0x2000, 0, 0, (rct_sprite *)peep);
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp]: Erm, wait, why did you do this?!

Comment (janisozaur) [reply src/openrct2/peep/Peep.cpp]: The function consuming this value checks for `x > 0x1fff` IIRC and `0x8000` is not a valid `sint16`

Comment (janisozaur) [reply src/openrct2/peep/Peep.cpp]: https://github.com/OpenRCT2/OpenRCT2/blob/d7f1f13d0c6e82e0e5795942e9c80d378d975b6a/src/openrct2/world/sprite.c#L541-L545

Comment (Gymnasiast) [reply src/openrct2/peep/Peep.cpp]: There is the SPRITE_LOCATION_NULL constant, which _is_ valid sint16.
I have tackled this in another PR.

Comment (janisozaur) [reply src/openrct2/peep/Peep.cpp]: ok, thanks

**Pull request merged**